File: 4.rb
def palindrome?(n)
  n.to_s == n.to_s.reverse
end

largest=0
r = (111..999)
r.each do |r1|
  r.each do |r2|
    p = r2*r1
    if palindrome?(p) and p > largest
      largest = p
    end
  end
end

puts largest

File: 04/4.exs
defmodule Palindrome do
  def is_palindrome?(num) do
    string = Integer.to_string(num)
    string == String.reverse(string)
  end

  def biggest_palindrome() do
    111..999
    |> Enum.flat_map(fn x -> Enum.map(x..999, fn y -> x * y end) end)
    |> Enum.filter(&is_palindrome?/1)
    |> Enum.max()
  end
end

IO.puts(Palindrome.biggest_palindrome())
